src/test/java/calculator/StringCalculatorTest.java

```diff
@@ -0,0 +1,54 @@
+package calculator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class StringCalculatorTest {
+
+    private StringCalculator cal;
+
+    @BeforeEach
+    void setUp() {
+        cal = new StringCalculator();
+    }
+
+    @Test
+    void add() {
+        String text = "1,2,3";
+        int result = cal.add(text);
+        assertThat(result).isEqualTo(6);
+    }
+
+    @Test
+    void add2() {
+        String text = "1:2,3";
+        int result = cal.add(text);
+        assertThat(result).isEqualTo(6);
+    }
+
+    @Test
+    void addCustomDelimiter() {
+        String text = "//;\n1;2;3";
+        int result = cal.add(text);
+        assertThat(result).isEqualTo(6);
+    }
+
+    @Test
+    void addException() {
+        String text = "1:2,-3";
+        assertThrows(RuntimeException.class, () -> cal.add(text));
+    }
+
+    @Test
+    void emptyString() {
+        String text = null;
+        String text2 = "";
+        int result = cal.add(text);
+        int result2 = cal.add(text2);
+        assertThat(result).isZero();
+        assertThat(result2).isZero();
+    }
```

Comment:
```suggestion
    @ParameterizedTest
    @NullAndEmptySource
    void validString(string text) {
        int result = cal.add(text);
        
        assertThat(result).isZero();
    }
```

emptyString() 의 경우 하나의 테스트 코드에 두 가지 경우(null, "") 를 테스트하고 있네요.

null과 "" 은 다른 경우라 nullString(), emptyString()으로 테스트 코드를 두 번 작성하거나
편의성을 위해 @ParameterizedTest 를 사용하는건 어떠신가요? 


https://www.baeldung.com/parameterized-tests-junit-5

Reply:
@ParameterizedTest 가 있었네요! 그리고 말씀하신 것처럼 null과 빈 문자열은 확실히 다른 것 같습니다 ㅎㅎ;;; 예시까지 들어주신@ParameterizedTest 사용을 해보도록 하겠습니다! 감사해용 ㅎㅎ